reco_utils/recommender/tfidf/tfidf_utils.py

```diff
@@ -0,0 +1,400 @@
+        
+        Args:
+            df_clean (pd.DataFrame): Dataframe with cleaned text in the new column.
+            text_col (str): Name of column containing the cleaned text.
+            ngram_range (tuple of int): The lower and upper boundary of the range of n-values for different n-grams to be extracted.
+            min_df (int): When building the vocabulary ignore terms that have a document frequency strictly lower than the given threshold.
+        
+        Returns:
+            tf (TfidfVectorizer): Scikit-learn TfidfVectorizer object defined in .tokenize_text().
+            vectors_tokenized (pd.Series): Each row contains tokens for respective documents separated by spaces.
+        """
+        vectors = df_clean[text_col]
+
+        # If a HuggingFace BERT word tokenization method
+        if self.tokenization_method in ['bert','scibert']:
+            # Set vectorizer
+            tf = TfidfVectorizer(analyzer='word', ngram_range=ngram_range, min_df=min_df, stop_words='english')
+
+            # Get appropriate transformer name
+            if self.tokenization_method == 'bert':
+                bert_method = 'bert-base-cased'
+            elif self.tokenization_method == 'scibert':
+                bert_method = 'allenai/scibert_scivocab_cased'
+            
+            # Load pre-trained model tokenizer (vocabulary)
+            tokenizer = BertTokenizer.from_pretrained(bert_method)
+            
+            # Loop through each item
+            vectors_tokenized = vectors.copy()
+            for i in range(0, len(vectors)):
+                vectors_tokenized[i] = ' '.join(tokenizer.tokenize(vectors[i]))
+
+        elif self.tokenization_method == 'nltk':
+            # NLTK Stemming
+            token_dict = {}
+            stemmer = PorterStemmer()
+
+            def stem_tokens(tokens, stemmer):
+                stemmed = []
+                for item in tokens:
+                    stemmed.append(stemmer.stem(item))
+                return stemmed
+
+            def tokenize(text):
+                tokens = nltk.word_tokenize(text)
+                stems = stem_tokens(tokens, stemmer)
+                return stems
+            
+            # When defining a custome tokenizer with TfidfVectorizer, the tokenization is applied in the fit function
+            tf = TfidfVectorizer(tokenizer=tokenize, analyzer='word', ngram_range=ngram_range, min_df=min_df, stop_words='english')
+            vectors_tokenized = vectors
+
+        elif self.tokenization_method == 'none':
+            # No tokenization applied
+            tf = TfidfVectorizer(analyzer='word', ngram_range=ngram_range, min_df=min_df, stop_words='english')
+            vectors_tokenized = vectors
+        
+        # Save to class variable
+        self.tf = tf
+
+        return tf, vectors_tokenized
+    
+    def fit_tfidf(self, tf, vectors_tokenized):
```

Comment:
suggest remove one-line wrapper function

Reply:
Even though this is just one line, I am deciding to leave this function in to follow the standard convention for models in this repo re: having a `fit()` function.